.devcontainer/devcontainer.json

```
@@ -0,0 +1,126 @@
+        "AZURE_BACKEND_ENDPOINT": "http://devstoreaccount1.blob.azure-mock.zenko.local",
+        "AZURE_BACKEND_QUEUE_ENDPOINT": "http://devstoreaccount1.queue.azure-mock.zenko.local",
+        "AZURE_SECRET_KEY": "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==",
+        "AWS_ENDPOINT": "http://aws-mock.zenko.local",
+        "AWS_ACCESS_KEY": "accessKey1",
+        "AWS_SECRET_KEY": "verySecretKey1",
+        "VERIFY_CERTIFICATES": "false",
+        // secrets-env
+        "AWS_BACKEND_SOURCE_LOCATION": "awsbackend",
+        "AWS_BACKEND_DESTINATION_LOCATION": "awsbackendmismatch",
+        "AWS_BACKEND_DESTINATION_FAIL_LOCATION": "awsbackendfail",
+        "GCP_BACKEND_DESTINATION_LOCATION": "gcpbackendmismatch",
+        "AZURE_BACKEND_DESTINATION_LOCATION": "azurebackendmismatch",
+        "COLD_BACKEND_DESTINATION_LOCATION": "e2e-cold",
+        "AZURE_ARCHIVE_BACKEND_DESTINATION_LOCATION": "e2e-azure-archive",
+        "LOCATION_QUOTA_BACKEND": "quotabackend",
+        "AWS_BUCKET_NAME": "ci-zenko-aws-target-bucket",
+        "AWS_CRR_BUCKET_NAME": "ci-zenko-aws-crr-target-bucket",
+        "AWS_FAIL_BUCKET_NAME": "ci-zenko-aws-fail-target-bucket",
+        "AZURE_CRR_BUCKET_NAME": "ci-zenko-azure-crr-target-bucket",
+        "AZURE_ARCHIVE_BUCKET_NAME": "ci-zenko-azure-archive-target-bucket",
+        "AZURE_ARCHIVE_BUCKET_NAME_2": "ci-zenko-azure-archive-target-bucket-2",
+        "AZURE_ARCHIVE_QUEUE_NAME": "ci-zenko-azure-archive-target-queue",
+        "GCP_CRR_BUCKET_NAME": "ci-zenko-gcp-crr-target-bucket",
+        "GCP_CRR_MPU_BUCKET_NAME": "ci-zenko-gcp-crr-mpu-bucket",
+        "ENABLE_RING_TESTS": "false",
+        "RING_S3C_BACKEND_SOURCE_LOCATION": "rings3cbackendingestion",
+        "RING_S3C_INGESTION_SRC_BUCKET_NAME": "ingestion-test-src-bucket-codespace-0",
+        "NOTIF_DEST_NAME": "destination1",
+        "NOTIF_DEST_TOPIC": "destination-topic-1",
+        "NOTIF_ALT_DEST_NAME": "destination2",
+        "NOTIF_ALT_DEST_TOPIC": "destination-topic-2",
+        "SUBDOMAIN": "zenko.local",
+        "SKOPEO_PATH": "/tmp",
+        "HELM_VERSION": "v3.5.3",
+        "YQ_VERSION": "v4.27.5",
+        "YQ_BINARY": "yq_linux_amd64",
+        "KUSTOMIZE_VERSION": "v4.4.1",
+        "GO_VERSION": "1.16.2",
+        "SKOPEO_VERSION": "v1.5.2",
+        "KUBECTL_VERSION": "1.21.1",
+        "TILT_VERSION": "0.23.4",
+        "KIND_VERSION": "v0.20.0",
+        "ZENKO_ENABLE_SOSAPI": "false",
+        "EXPIRE_ONE_DAY_EARLIER": "true",
+        "TRANSITION_ONE_DAY_EARLIER": "true",
+        "TIME_PROGRESSION_FACTOR": "1",
+        "ZENKO_MONGODB_DATABASE": "zenko-database",
+        "ZENKO_MONGODB_SHARDED": "true" 
+        },
+
+    // Use 'forwardPorts' to make a list of ports inside the container available locally.
+    // "forwardPorts": [],
+
+    // Use 'postCreateCommand' to run commands after the container is created.
+    "postCreateCommand": "bash .devcontainer/setup.sh"
```

While this may become useless in the near future, let's keep it while we are doing the `docker login` in the `setup.sh`

```suggestion
    "secrets": {
        "REGISTRY_LOGIN": {
            "description": "Login registry."
        },
        "REGISTRY_PASSWORD": {
            "description": "Password registry."
        }
    },
    // Use 'postCreateCommand' to run commands after the container is created.
    "postCreateCommand": "bash .devcontainer/setup.sh"
```